### ISSUE TITLE
Unclear syntax for contentSize values

### ISSUE THREAD

**Issue by dmejia** [schemaorg/schemaorg#869]
http://schema.org/contentSize is a Text field documented as "File size in (mega/kilo) bytes."
What's the expected syntax for this text? Should it include the units and if so what should be the format?
i.e "42.54MB" vs "42.54 megabytes" vs "42.54" ...

Thanks!

**Comment by mfhepp** [comment]
I think we should add schema:QuantitativeValue to the range of this property.  There are UN/CEFACT Common Codes for all major units of measurement, like

Byte / Octet - code: Q12
kB / 10³ bytes - code: 2P
MB / 10⁶ bytes - code: 4L  
GB - code: E34
TB - code: E35 

Martin

**Comment by mfhepp** [comment]
It would also help to add an example with schema:QuantitativeValue once this addition has been made.

**Comment by danbri** [comment]
Thanks. Yes - there is a general need to document our abbreviated textual form for such measurements. In other properties (which?) we accept "10 m" without saying whether the "m" is (obviously) metres or miles. Having a short form is fine, but we need to be clearer what the form is short *for*.

(@mfhepp and I spoke recently about possibly collecting some test cases in this direction...)

**Comment by dmejia** [comment]
Do QuantitativeValues support abbreviated textual forms? From the docs I thought they (specifically StructuredValues) were meant for returning structured information instead of text. 

For instance http://schema.org/width can be a QuantitativeValue but also a "Distance." Distance does specify the expected format <number> <length unit>. Should contentSize be consistent with that? (i.e support QuantitativeValue in addition to some DigitalSize textual format) Or QuantitativeValues are supposed to replace that and support textual forms?

Note that the same (what's used for distances) isn't true of something like http://schema.org/weight though which is just defined as a QuantitaveValue. I'm not sure I like that though because if everything is defined as QuantitativeValues it won't be possible to check for instance which schemas have weights, or lengths, etc, and the correct type is basically not enforced (I'd leave that for a separate issue though, first I'd like to understand the currently proper way to use contentSize).

**Comment by mfhepp** [comment]
The QuantitativeValue mechanism is agnostic of any further distinction of the type of value, like distance, weight, mass, density of matter, etc. To my knowledge, the schema:Distance type in schema.org is a remainder from the beginning of schema.org where a conceptually cleaner separation of quantities was considered.

You are right that you cannot currently express that a particular property with a range of schema:QuantitativeValue accepts only values of a certain type or with a small subset of units of measurement. Instead, we try to indicate typical unit codes in the description of the property.

Since the set of potential units of measurement and types of quanties is huge, and since Web masters are facing difficulties changing their markup based on the type of a quantitative value, the current approach is, imo, better.

Note that, in general, schema.org does not formulate a lot of integrity constraints in the vocabulary.

Martin
-----------------------------------
martin hepp  http://www.heppnetz.de
mhepp@computer.org          @mfhepp

**Comment by csarven** [comment]
Would it be reasonable to suggest using datatype with the value instead of unit being included in the value?

**Comment by csarven** [comment]
What's the difference between schema:contentSize and schema:fileSize?

**Comment by csarven** [comment]
Alternatively, update the definition by borrowing fileSize's definition ie. "In the absence of a unit (MB, KB etc.), KB will be assumed."

**Comment by bmhm** [comment]
This should not be stale.
I suggest using the kubernetes convention:

> Meaning of memory
> Limits and requests for memory are measured in bytes. You can express memory as a plain integer or as a fixed-point integer using one of these suffixes: E, P, T, G, M, K. You can also use the power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki. For example, the following represent roughly the same value:

Source: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

**Comment by jonoalderson** [comment]
Chipping in here that it'd be _really_ nice if this would also accept a `QuantitativeValue`, rather than just `text`.

**Comment by bmarwell** [comment]
Hello, this is not resolved...

**Comment by bmarwell** [comment]
Hello, this is not resolved...

**Comment by bmarwell** [comment]
Yeah maybe after 10 years just leave it undefined and therefore defunctional. Seems to work for most of us, otherwise this issue would have got more traction. ;)
